microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/project/JDTMicroProfileProject.java

```diff
@@ -45,23 +48,89 @@ public JDTMicroProfileProject(IJavaProject javaProject) {
 		configSources.add(new PropertiesConfigSource(MICROPROFILE_CONFIG_PROPERTIES_FILE, javaProject));
 	}
 
-	public String getProperty(String key, String defaultValue) {
-		for (IConfigSource configSource : configSources) {
-			String property = configSource.getProperty(key);
-			if (property != null) {
-				return property;
+	/**
+	 * Returns the value of this property or <code>defaultValue</code> if it is not
+	 * defined in this project
+	 *
+	 * @param propertyKey  the property to get with the profile included, in the
+	 *                     format used by microprofile-config.properties
+	 * @param defaultValue the value to return if the value for the property is not
+	 *                     defined in this project
+	 * @return the value of this property or <code>defaultValue</code> if it is not
+	 *         defined in this project
+	 */
+	public String getProperty(String propertyKey, String defaultValue) {
+		// Need to enforce the order that config files are looked at
+		// eg. prefer a Yaml config value over application.properties
+		IConfigSource configSource;
+		for (int i = 0; i < configSources.size(); i++) {

```

Comment:
`for(IConfigSource configSource : configSources) {`

Reply:
I needed to change to iterating backwards through the List of config sources, and it looks like using indexing is the cleanest method.

Reply:
I don't understand, `for(IConfigSource configSource : configSources) {` is the same thing than `for (int i = 0; i < configSources.size(); i++) {`, no?

Reply:
Oops I forgot to push